### PR TITLE
Add IO and Statistics User Guide

### PR DESCRIPTION
**Pull request by dallasfoster**
# Earth2Studio Pull Request

## Description
Adds user guides documentation for IO backends, Statistics, and Metrics.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: docs/userguide/features/statistics.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+# Metrics
+
+Like statistics, metrics are reductions across existing dimensions. Unlike statistics,
+which are usually defined over a single input, we define metrics to take a pair of
+inputs. Otherwise, the API and requirements are similar to the statistics requirements.
+
+## Metrics Interface
+
+```{literalinclude} ../../../earth2studio/io/base.py
```
Comment (NickGeneva) [inline docs/userguide/features/statistics.md]: Update file

File: docs/userguide/features/statistics.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+we assume that statistics reduce existing coordinates so that the output tensors
+have a coordinate system that is a subset of the input coordinate system. This
+makes statistics less flexible than diagnostic models but have fewer API requirements.
+
+## Statistics Interface
+
+Statistics API only specify a {func}`__call__` method that matches similar methods
+across the package.
+
+```{literalinclude} ../../../earth2studio/io/base.py
```
Comment (NickGeneva) [inline docs/userguide/features/statistics.md]: does this need update

Comment (dallasfoster) [reply docs/userguide/features/statistics.md]: updated

File: docs/userguide/features/statistics.md (hunk 1 above)
Comment (dallasfoster) [reply docs/userguide/features/statistics.md]: updated.

File: docs/userguide/features/io.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+# Output Handling
+
+While input data handling is primarily managed by the data sources in
+{ref}`earth2studio.data`, output handling is managed by the IO backends available
+in {ref}`earthstudio.io`. These backends are designed to balance the ability for
+users to customize the arrays and metadata within the exposed backend while also
+making it easy to design resuable workflows.
+
+The key extension of the typical `(x, coords)` data structure movement throughout
+the rest of the `earthstudio` code and output store compatibility is the notion of
```
Comment (NickGeneva) [inline docs/userguide/features/io.md]: "earth2studio"

Comment (dallasfoster) [reply docs/userguide/features/io.md]: updated this and a few others.

**Pull request merged**